rust/parquet/src/arrow/record_reader.rs

```diff
@@ -0,0 +1,674 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::cmp::{max, min};
+use std::mem::replace;
+use std::mem::size_of;
+use std::mem::transmute;
+use std::slice;
+
+use crate::column::{page::PageReader, reader::ColumnReaderImpl};
+use crate::data_type::DataType;
+use crate::errors::{ParquetError, Result};
+use crate::schema::types::ColumnDescPtr;
+use arrow::bitmap::Bitmap;
+use arrow::buffer::{Buffer, MutableBuffer};
+use arrow::builder::{BooleanBufferBuilder, BufferBuilderTrait};
+
+const MIN_BATCH_SIZE: usize = 1024;
+
+/// A `RecordReader` is a stateful column reader that delimits semantic records.
+pub struct RecordReader<T: DataType> {
+    column_desc: ColumnDescPtr,
+
+    records: MutableBuffer,
+    def_levels: Option<MutableBuffer>,
+    rep_levels: Option<MutableBuffer>,
+    null_bitmap: Option<BooleanBufferBuilder>,
+    column_reader: Option<ColumnReaderImpl<T>>,
+
+    /// Number of records accumulated in records
+    records_num: usize,
+
+    /// Last offset of `records_num` records
+    values_pos: usize,
```

Comment:
How is this used? seems it is only assigned.

Reply:
It was used in previous version of this algorithm, remove this now.